### PR TITLE
Text alignment and multiline support

### LINKED ISSUE #594: UiText alignment
Feature request

Align left default----
--align center--
----align right

Also, if the text doesn't fit into the mesh holding it, it might be a good idea to have an "overflow" option.
if we reverse the problem, changing the text size so it fits onto the mesh could be another option.

Also top,center and bottom vertical alignements.
When I use my layouting PR to set the text to be fill a rectangle, having the text centered in the middle would be nice ;)
Less testing required to align stuff this way.

Since this requires modifying the pass, I'd consider this average difficulty.

### PR DESCRIPTION
**Pull request by jojolepro**
Fixes #594
Fixes https://github.com/amethyst/amethyst/issues/928

image

### CODE REVIEW

File: amethyst_ui/src/font/systemfont.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+use font_kit::handle::Handle;
+use font_kit::error::SelectionError;
+use font_kit::source::SystemSource;
+use font_kit::family_name::FamilyName;
+use font_kit::properties::{Properties, Style};
+
+// Temporarily disabled because const can't contain heap allocations.
+// Fantasy contains comic sans btw
+//const DEFAULT_FONTS: &[FamilyName] = &[FamilyName::Title("arial".to_string()), FamilyName::SansSerif, FamilyName::Monospace, FamilyName::Fantasy];
```
Comment (Moxinilian) [inline amethyst_ui/src/font/systemfont.rs]: Not interfering with the merge, but I'd recommend PRing Arial as a default family name to that crate you use.

**Comment by jojolepro** [comment]
image

**Comment by Moxinilian** [comment]
Beautiful

File: amethyst_ui/src/font/systemfont.rs (hunk 1 above)
Comment (jojolepro) [reply amethyst_ui/src/font/systemfont.rs]: I would actually PR a way to declare FamilyName without an allocation if possible

Comment (Xaeroxe) [reply amethyst_ui/src/font/systemfont.rs]: Should remove this, const will never have the ability to contain heap allocations.

**Comment by jojolepro** [comment]
Ready to go.
There's a bug where if you do the following input sequence on an input text, a char can become invisible. I don't know where the bug comes from and didn't see anything that looked suspicious on the amethyst side (I checked text.rs line 524).

```
select edit text
erase all text
press a
press backspace
press a (invisible char)
press a (both "a" are visible)
Works with any char key, not just "a"
```

**Pull request merged**